Development

PhMueller opened the pull request:
Upgrade the master to the version.

KEggensperger commented:
@mfeurer This is not yet ready, correct? Could you ping me once you think this is ready to be merged?

mfeurer commented on hpolib/benchmarks/ml/autosklearn_benchmark.py:
```diff
@@ -85,10 +111,10 @@ def _get_data_manager(self, task_id):
             # TODO in the future, the XYDataManager should have this as it's own
```
It's actually solved, I just didn't see the todo.

mfeurer commented on optional-requirements.txt:
```diff
@@ -9,8 +9,8 @@ scikit-learn>=0.19.0,<0.20.0
 
 # for benchmarks using auto-sklearn
-auto-sklearn==0.3.0
 pynisher
```
Good point, will do.

mfeurer commented on requirements.txt:
```diff
@@ -6,5 +6,5 @@ scikit-learn>=0.19.0,<0.20.0
 liac-arff>=2.1.1
 ConfigSpace>=0.4.0
 openml
-pynisher
```
Good point. We only need it for testing and it is specified in the test requirements. I will remove it.

KEggensperger commented on hpolib/benchmarks/ml/autosklearn_benchmark.py:
```diff
@@ -30,22 +30,48 @@ class AutoSklearnBenchmark(AbstractBenchmark):
     auto-sklearn benchmarks implement Section 6 of the paper 'Efficient and
     Robust Automated Machine Learning' by Feurer et al., published in
     Proceedings of NIPS 2015.
+
```
Shouldn'T this comment be inside __init__?

mfeurer replied:
I'm always unsure where such comments should be, but basically, this is about the class itself, not about the init method.

KEggensperger commented on hpolib/benchmarks/ml/sklearn_mlp_benchmark.py:
```diff
@@ -175,9 +175,3 @@ def get_meta_information():
                     "publisher={IEEE}"]
         d["references"].append(data_ref)
         return d
-
```
Is removing the benchmark really part of this PR?

KEggensperger commented:
@mfeurer Can you have a look at the error message on Travis? If this is solved, we can finally merge ...

```
ERROR: test_load_openml (test_openml_data_manager.TestDataManager)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/automl/HPOlib2/tests/test_openml_data_manager.py", line 10, in test_load_openml
    X_train, y_train, X_val, y_val, X_test, y_test = dm.load()
  File "/home/travis/build/automl/HPOlib2/hpolib/util/openml_data_manager.py", line 94, in load
    self.task_id)
  File "/home/travis/build/automl/HPOlib2/hpolib/util/openml_data_manager.py", line 14, in _load_data
    task = openml.tasks.get_task(task_id)
  File "/home/travis/miniconda/envs/testenv/lib/python3.5/site-packages/openml/tasks/functions.py", line 310, in get_task
    raise e
  File "/home/travis/miniconda/envs/testenv/lib/python3.5/site-packages/openml/tasks/functions.py", line 301, in get_task
    task.download_split()
  File "/home/travis/miniconda/envs/testenv/lib/python3.5/site-packages/openml/tasks/task.py", line 116, in download_split
    split = OpenMLSplit._from_arff_file(cached_split_file)
  File "/home/travis/miniconda/envs/testenv/lib/python3.5/site-packages/openml/tasks/split.py", line 90, in _from_arff_file
    type_idx = meta._attrnames.index('type')
AttributeError: 'MetaData' object has no attribute '_attrnames'
```

mfeurer commented:
That's actually an issue with OpenML at the moment, not sure what happened there. I can check this after Friday.

Commit:
init 0.0.5 (#42)

PhMueller committed:
PyBNN Benchmark (#43)

* [skip travis] Initial commit for the pyBNN

* [skip travis] PyBNN: create DataManager + make Benchmark deterministic

* [skip travis] PyBNN singularity recipe

* PyBNN tests

* ADD reference to BOHB

* FIX in tests

the tests take more time on travis. Check if the cost of a benchmark is simply > 1. (it should not be 0 or None)

* FIX in tests

- add availability test - check if we can find the container in the library.
- add meta data to pybnn and cartpole.py
- remove currently the nasbench201 tests

* FIX Pytest8 error

PhMueller committed:
ADD 0-1 scaling to svm features (#47)

Commit:
Fix pybnn/nasbench201epochs (#56)

* FIX using default theano compiledir
* FIX epochs [0,199] to [1,200]
* FIX lowest fidelity to be 500
* FIX codestyle errors
Co-authored-by: PhMueller <muller-phil@gmx.net>

PhMueller committed:
Bug54 testerror (#58)

* ADD 0-1 scaling to svm features

* Fix NasBench101 and tabular_benchmarks: They return now the test instead of the validation error.

* Update Tabular Benchmarks

* Update Nasbench201

* Update Version + Remove test code

* Small fix nasbench 201

* Small fix tabularbenchmarks

PhMueller committed:
Rename HPOlib -> HPOBench, hpolib -> hpobench (#61)

* Rename HPOlib -> HPOBench, hpolib -> hpobench

* Fix Tests

* Fix Tests

* Fix Tests

* Fix Example

PhMueller committed:
Merge branch 'master' into development

KEggensperger commented on README.md:
```diff
@@ -1,16 +1,16 @@
+**Note:** HPOBench is under active construction. Stay tuned for more benchmarks. Information on how to contribute a new benchmark will follow shortly.
 
-**Note:** If you are looking for a different version of HPOlib2, you might be looking for [HPOlib1.5](https://github.com/automl/HPOlib1.5) 
+**Note:** If you are looking for a different version of HPOBench, you might be looking for [HPOlib1.5](https://github.com/automl/HPOlib1.5) 
```
Replace with "If you are looking for a different or older version of our benchmarking library, you might be looking for ..."

KEggensperger commented on README.md:
````diff
@@ -78,29 +78,29 @@ pip install .
 
 ```bash
-$XDG_CONFIG_HOME # ~/.config/hpolib2
-$XDG_CACHE_HOME # ~/.cache/hpolib2
````
Sorry, I replaced this directly on master. There should be 3 different directories.

KEggensperger commented on hpobench/benchmarks/nas/nasbench_201.py:
```diff
@@ -281,10 +289,14 @@ def objective_function_test(self, configuration: Union[CS.Configuration, Dict],
         # The result dict should contain already all necessary information -> Just swap the function value from valid
         # to test and the corresponding time cost
-        result = self.objective_function(configuration=configuration, fidelity=fidelity, data_seed=(777, 888, 999),
+        assert fidelity['epoch'] == 199, 'Only test data for the 200. epoch is available. ' \
```
Shouldn't this be 200?

KEggensperger commented on hpobench/benchmarks/nas/tabular_benchmarks.py:
```diff
@@ -37,10 +37,10 @@
+from hpobench.abstract_benchmark import AbstractBenchmark
 
-__version__ = '0.0.1'
+__version__ = '0.0.3'
```
Shouldn't this rather be 0.0.2?

KEggensperger commented on changelog.md:
```diff
@@ -1,3 +1,6 @@
+# 0.0.5
+  * Rename package to HPOBench
+
```
There is probably a bit more:
  * Added BNN benchmark
  * Updated returned loss values for NB201

KEggensperger commented on hpobench/__version__.py:
```diff
@@ -0,0 +1 @@
+__version__ = '0.0.5dev'
```
This should be only "0.0.5"

PhMueller replied:
Thought about this too. But if I change it here, then the dev branch loses its _dev tag. So I thought, that we have to replace it directly on the master later, or is there another simple way I don't see? :-)

KEggensperger replied:
No, we do it here and then directly push a new commit to dev initializing the new version (see here: https://github.com/automl/HPOBench/commit/6ffe1a44d233d57aadee2624acc60cc27dc1c3ec)

KEggensperger reviewed (changes requested):
Thank you. There are a few minor things, but overall this looks good and can be merged.

PhMueller committed:
Integrate requested changes
